src/services/user-service/UserModel.ts

```diff
@@ -0,0 +1,5 @@
+import { UserInfoT } from './UserService';
```

Comment:
Maintain a contract between services and models.
It should be an interface instead of abstract class, however typescript interface cannot be DI. I use abstract class instead.